subprojects/core/src/test/groovy/org/gradle/api/internal/file/collections/DefaultConfigurableFileTreeTest.groovy

```diff
@@ -64,6 +64,7 @@ class DefaultConfigurableFileTreeTest extends AbstractTestForPatternSet {
         super.setUp()
         NativeServicesTestFixture.initialize()
         fileSet = new DefaultConfigurableFileTree(testDir, fileResolverStub, taskResolverStub, fileCopier, directoryFileTreeFactory())
+        fileCopier = new FileCopier(DirectInstantiator.INSTANCE, fileResolverStub, context.mock(FileLookup), new DefaultDirectoryFileTreeFactory())
```

Comment:
We should use `TestFiles.directoryTreeFactory()` in tests instead of creating new instances. There are a bunch of other tests doing this in this PR as well.